client/src/features/checkout/components/CheckoutForm/CheckoutForm.js

```diff
@@ -1,63 +1,82 @@
 /* eslint-disable no-unused-vars */
-import React, { useState } from 'react';
+import React from 'react';
 // Form components
 import { Formik, Form } from 'formik';
 import { LocalizationProvider } from '@mui/x-date-pickers';
 import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
 import { enAU } from 'date-fns/locale';
 // MUI Components
 import { Stepper, Step, StepLabel, Button, CircularProgress, Box } from '@mui/material';
+// Redux store
+import { useSelector, useDispatch, shallowEqual } from 'react-redux';
+import {
+  increaseStep,
+  decreaseStep,
+  createNewOrder,
+  setIsModalOpen,
+} from '../../../../store/slices/orderSlice/orderSlice';
 // Child Forms and model
 import { UserDetailsForm, ShippingAddressForm, PaymentForm, PaymentSuccess, CheckoutSummary } from '..';
 import { initialValues, validationSchema } from '../../data';
 
 const steps = ['User Details', 'Shipping Address', 'Payment Details'];
 
 const CheckoutForm = () => {
-  const [activeStep, setActiveStep] = useState(0);
+  const { currentStep, isLoading } = useSelector((state) => state.order, shallowEqual);
```

Comment:
i guess shallowEqual is only used for complex data types like arrays and objects which is not our case here

Reply:
already discussed